apps/admin_app/assets/js/components/CreatePromotions.tsx

```diff
@@ -0,0 +1,369 @@
+            expires_at:"",
+            usageLimit:"",
+            usageCount:"Nil",
+            matchPolicy:"",
+            activeStatus:false,
+            success:false,
+            ruleDown:false,
+            addRule:false,
+            actiondown:false,
+            addedRules:[],
+            availableRules:"",
+            addAction:false,
+            availableActions:"",
+            availableCalculators:"",
+            selectedRule:"",
+            selectedCalc:"",
+            lowRange:"",
+            upRange:"",
+            selectedProducts:"",
+            productMatchPolicy:"",
+            addedActions:[],
+            selectedAction:"",
+            selectedCalculator:""
+        }
+
+        this.handleStartDate = this.handleStartDate.bind(this);
+        this.handleEndDate = this.handleEndDate.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+        this.handleAddRule = this.handleAddRule.bind(this); 
+        this.fetchPOST = this.fetchPOST.bind(this);
+        this.handleAddAction = this.handleAddAction.bind(this);
+        this.handleRule = this.handleRule.bind(this);
+        this.renderRuleOptions = this.renderRuleOptions.bind(this);
+        this.handleCalculator = this.handleCalculator.bind(this);
+        this.renderCalculatorOptions = this.renderCalculatorOptions.bind(this);
+        this.handleSaveRule = this.handleSaveRule.bind(this);
+        this.onSuccess = this.onSuccess.bind(this);
+        this.handleDeleteRule = this.handleDeleteRule.bind(this);
+    }
+
+    handleStartDate = (date,dateString)=>{
+        const startDate = new Date(dateString)
+        startDate.setHours(0,0,0)
+        this.setState({starts_at:startDate})
+    }
+
+    handleEndDate = (date,dateString)=>{
+        const endDate = new Date(dateString)
+        endDate.setHours(0,0,0)
+        this.setState({expires_at:endDate})
+    }
+
+    fetchPOST = (url,data)=>{
+        return fetch(url,{
+            credentials: 'include',
+            method:'POST',
+            body:JSON.stringify(data),
+            headers: {'Content-Type': 'application/json; charset=UTF-8'},
+        }).then(res =>res.json())
+        
+    }
+
+    fetchGET =(url)=>{
+        return fetch(url,{
+            credentials:'include',
+            method:"GET",
+            headers:{'Content-Type': 'application/json; charset=UTF-8'}
+        }).then(res =>res.json())
+    }
+
+    handleSave = () =>{
+        var data = {
+            "data":{
+                "starta_at":this.state.starts_at,
+                "expires_at":this.state.expires_at,
+                "code":this.state.code,
+                "name":this.state.name,
+                "rules":[],
+                "actions":[]
+            }
+        }
+        var url = 'http://localhost:4000/api/promotions'
+        this.fetchPOST(url,data).then(response=>{
+            var arrofkeys=Object.keys(response)
+            arrofkeys[0]=="errors"?this.onFail(response):this.onSuccess()
```

Comment:
string comparison is ideally not the right way. We should think of something else here.